api/provisioning/v1alpha1/provisioningrequest_types.go

```diff
@@ -93,6 +93,39 @@ type PolicyDetails struct {
 	RemediationAction string `json:"remediationAction,omitempty"`
 }
 
+// ProvisioningState defines the various states of the provisioning process.
+type ProvisioningState string
+
+const (
+	// StateProgressing means the provisioning process is currently in progress.
+	// It could be in progress during hardware provisioning, cluster installation, or cluster configuration.
+	StateProgressing ProvisioningState = "progressing"
+
+	// StateFulfilled means the provisioning process has been successfully completed for all stages.
+	StateFulfilled ProvisioningState = "fulfilled"
+
+	// StateFailed means the provisioning process has failed at any stage, including resource validation
+	// and preparation prior to provisioning, hardware provisioning, cluster installation, or cluster configuration.
+	StateFailed ProvisioningState = "failed"
+)
+
+// ProvisionedResources contains the resources that were provisioned as part of the provisioning process.
+type ProvisionedResources struct {
+	// The identifier of the provisioned oCloud Cluster.
+	OCloudNodeClusterId string `json:"oCloudNodeClusterId,omitempty"`
+}
+
+type ProvisioningStatus struct {
+	// The current state of the provisioning process.
+	// +kubebuilder:validation:Enum=progressing;fulfilled;failed
+	ProvisioningState ProvisioningState `json:"provisioningState,omitempty"`
+	// The detail about the current state of the provisioning process.
+	ProvisioningDetail string `json:"provisioningDetail,omitempty"`
```

Comment:
I think that `ProvisioningDetails` would make more sense, even if there's just one piece of extra information, but if this is the requirement, then I guess it's ok.

Reply:
I think it is OK to update this to `ProvisioningDetails` - I don't believe we agreed to naming at this level and if we think plural makes sense then lets change it now.

Reply:
No requirement for this name. Yeah, I used `ProvisioningDetail` as I thought only one item for this. Okay, I can change to `ProvisioningDetails` if that makes more sense.